tools/serverpod_cli/test/generator/dart/server_code_generator/sealed_class_test.dart

```diff
@@ -750,24 +750,23 @@ void main() {
       config: config,
     );
 
-    var grandparentCompilationUnit = parseString(
-      content: codeMap[getExpectedFilePath(
-        grandparent.fileName,
-        subDirParts: ['sub_dir'],
-      )]!,
-    ).unit;
-    var parentCompilationUnit =
-        parseString(content: codeMap[getExpectedFilePath(parent.fileName)]!)
-            .unit;
-    var childCompilationUnit =
-        parseString(content: codeMap[getExpectedFilePath(child.fileName)]!)
-            .unit;
+    var grandparentPath = getExpectedFilePath(
+      grandparent.fileName,
+      subDirParts: ['sub_dir'],
+    );
+    var parentPath = getExpectedFilePath(parent.fileName);
+    var childPath = getExpectedFilePath(child.fileName);
+
+    var grandparentCompilationUnit =
+        parseString(content: codeMap[grandparentPath]!).unit;
+    var parentCompilationUnit = parseString(content: codeMap[parentPath]!).unit;
+    var childCompilationUnit = parseString(content: codeMap[childPath]!).unit;
 
     group('then ${grandparent.className}', () {
       test('has a part directive with ${parent.className} uri', () {
         var partDirective = CompilationUnitHelpers.tryFindPartDirective(
           grandparentCompilationUnit,
-          uri: '../${parent.fileName}.dart',
+          uri: p.relative(parentPath, from: p.dirname(grandparentPath)),
```

Comment:
Question(looking at the commit that introduced this): This is where we introduce the windows bug right? When the path is no longer "dart" compatible.

Reply:
No the tests on the commit before failed too. 

But according to [this comment on the bug report](https://github.com/dart-lang/sdk/issues/59629#issuecomment-2512094587) it should have already worked with `../${parent.fileName}.dart`.